lib/vrl/stdlib/src/chunks.rs

```diff
@@ -0,0 +1,221 @@
+            },
+            Parameter {
+                keyword: "utf8",
+                kind: kind::BOOLEAN,
+                required: true,
+            },
+        ]
+    }
+
+    fn examples(&self) -> &'static [Example] {
+        &[
+            Example {
+                title: "chunks by byte when utf8=true",
+                source: r#"chunks("abcdefgh", 4, true)"#,
+                result: Ok(r#"["abcd", "efgh"]"#),
+            },
+            Example {
+                title: "chunk sizes respect unicode code point boundaries when utf8=true",
+                source: r#"chunks("ab你好", 4, true)"#,
+                result: Ok(r#"["ab", "你", "好"]"#),
+            },
+            Example {
+                title: "chunk size cannot be less than 4 bytes when utf8=true",
+                source: r#"chunks("hello world", 3, true)"#,
+                result: Err(
+                    r#"function call error for "chunks" at (0:30): "chunk_size" must be at least 4 bytes"#,
+                ),
+            },
+            Example {
+                title: "chunk size can be less than 4 bytes when utf8=false",
+                source: r#"chunks("hello world", 3, false)"#,
+                result: Ok(r#"["hel","lo ","wor","ld"]"#),
+            },
+            Example {
+                title: "chunk size must be at least 1 byte",
+                source: r#"chunks("hello world", 0, false)"#,
+                result: Err(
+                    r#"function call error for "chunks" at (0:31): "chunk_size" must be at least 1 byte"#,
+                ),
+            },
+            Example {
+                title: "chunk sizes do not respect unicode code point boundaries when utf8=false",
+                source: r#"chunks("ab你好", 4, false)"#,
+                result: Ok(r#"["ab�","�好"]"#),
+            },
+        ]
+    }
+
+    fn compile(
+        &self,
+        _state: (&mut state::LocalEnv, &mut state::ExternalEnv),
+        _ctx: &mut FunctionCompileContext,
+        mut arguments: ArgumentList,
+    ) -> Compiled {
+        let value = arguments.required("value");
+        let chunk_size = arguments.required("chunk_size");
+        let utf8 = arguments.required("utf8");
+
+        Ok(Box::new(ChunksFn {
+            value,
+            chunk_size,
+            utf8,
+        }))
+    }
+}
+
+#[derive(Debug, Clone)]
+struct ChunksFn {
+    value: Box<dyn Expression>,
+    chunk_size: Box<dyn Expression>,
+    utf8: Box<dyn Expression>,
+}
+
+impl Expression for ChunksFn {
+    fn resolve(&self, ctx: &mut Context) -> Resolved {
+        let value = self.value.resolve(ctx)?;
+        let chunk_size = self.chunk_size.resolve(ctx)?;
+        let utf8 = self.utf8.resolve(ctx)?;
+
+        chunks(value, chunk_size, utf8)
+    }
+
+    fn type_def(&self, _state: (&state::LocalEnv, &state::ExternalEnv)) -> TypeDef {
+        TypeDef::array(Collection::from_unknown(Kind::bytes())).infallible()
+    }
+}
+
+#[cfg(test)]
+mod tests {
```

Comment:
In addition to these tests, we usually add benchmarks for each new VRL function, as well, in `benches/benches.rs`.